py-rattler/src/version/mod.rs

```diff
@@ -132,10 +132,35 @@ impl PyVersion {
         }
     }
 
-    /// Returns a new version where the last numerical segment of this version has been bumped.
-    pub fn bump(&self) -> Self {
-        Self {
-            inner: self.inner.bump(),
+    /// Returns a new version where the major segment of this version has been bumped.
+    pub fn bump_major(&self) -> PyResult<Self> {
```

Comment:
If we had that integer option we could pass an int from Python (with -1 being the last element, for example), and bump like that. 
Then in Python we could also define an enum for the different named segments and allow both, passing an enum or an int.

Reply:
I am fine with Python API.

What we have so far is:

- `bump_major()`
- `bump_minor()`
- `bump_patch()`
- `bump_last()`

I can add the below as well:

- `bump_segment(int)`
- `bump(enum)`

Does that sound good to you?

Reply:
Python side done at https://github.com/mamba-org/rattler/pull/452/commits/4ac1858337f0039d950dcda5408caea378abe917

I didn't implement `bump(enum)` since it seems a bit overkill at this point, but let me know if you still want it.